### DISCUSSION TITLE
A roadmap for v1.1 and beyond

### DISCUSSION THREAD

**Discussion by 2zqa** [MCPHackers/RetroMCP-Java#119] [Ideas]
Continuing from https://github.com/MCPHackers/RetroMCP-Java/issues/118:

Development on this project has become stale, with only bug fixes and extra things being accepted if they are made in a pull request. It is perhaps interesting for potential new developers to know what bug fixes or features are deemed most useful for this project to get a new release.

If fleshed out properly, the GitHub Projects or the milestones feature could be used to track progress.

**Comment by PhoenixVX** [comment]
I'll list out a few things that would be nice but are no means necessary for a new release.

- [ ] Native support for ARM-based architectures (M1/M2/M3 macbooks, Raspberry Pis, etc)
- [ ] Merged decompilation with separate source sets (currently, merged decompilation decompiles into 1 entire source set)
- [x] Working Intellij support
- [x] VSCode run configs (Importing a project from the Eclipse configs doesn't work as intended so this is necessary)
- [ ] Gradle support (yeah...)
- [ ] More intuitive project structure (the current one sucks, we should've kept the old one)
- [ ] Layered mappings (half supported already... needs some work especially when it comes to remapping and merging JAR strategies)
- [ ] Better plugin support (The API already exists but it is rather unused and undocumented)
- [ ] Rewritten documentation pages (might be better to put this on the website instead though)
- [ ] Support for other JVM languages such as Kotlin and Scala
- [x] Fixing working directory not being read on startup (without making it jank af)
- [x] Codebase separation and better documentation of the codebase
- [ ] Improvements to RetroDebugInjector (RDI) which is used as the remapper / class fixer in RMCP (issues can be found in the issues page here)
- [ ] More versions added to the website (We do have mappings for a lot of versions which aren't on the website yet simply due to how time consuming it is to go through each version and create the proper files necessary to decompile)

**Comment by PhoenixVX** [comment]
Reusing this discussion for v1.1 and beyond, most of the features not checked here are still planned/wanting to be implemented.
